testsuite/test-runner/src/main/java/org/wildfly/core/testrunner/Server.java

```diff
@@ -224,11 +207,13 @@ protected void stop() {
                 }
 
                 if (process != null) {
-                    process.waitFor();
-                    process = null;
+                    if (process.waitFor(stopTimeout, TimeUnit.SECONDS)) {
+                        process = null;
+                    } else {
+                        // The process hasn't shutdown within the specified timeout. Terminate forcibly.
+                        process.destroy();
```

Comment:
The old logic amounted to

process.waitFor(timeout); // our own equivalent
if (!stopped) {
  process.destroy();
  process.waitFor();
}

Now it's

process.waitFor(timeout); // our own equivalent
if (!stopped) {
  process.destroy();
}

I don't think the second process.waitFor() should be dropped.

Reply:
Once the process.waitFor() is restored the process field must be nulled in all cases.

Reply:
@bstansberry Sorry, I don't understand what you mean by the second process.waitFor(). Current implementation starts a new thread that does the timeout loop. Original thread is blocked in process.waitFor(). If the timeout runs out process.destroy() is called which causes process.waitFor() in the original thread to return. Am I missing something?